common/src/main/java/org/opensearch/sdk/DeleteDataObjectResponse.java

```diff
@@ -0,0 +1,141 @@
+import org.opensearch.core.common.Strings;
+import org.opensearch.core.index.shard.ShardId;
+
+public class DeleteDataObjectResponse {
+    private final String id;
+    private final ShardId shardId;
+    private final ShardInfo shardInfo;
+    private final boolean deleted;
+
+    /**
+     * Instantiate this request.
+     * <p>
+     * For data storage implementations other than OpenSearch, an index may be referred to as a table and the id may be referred to as a primary key.
+     * @param id the document id
+     * @param shardId the shard id
+     * @param shardInfo the shard info
+     * @param deleted Whether the object was deleted. Use {@code false} if the object was not found.
+     */
+    public DeleteDataObjectResponse(String id, ShardId shardId, ShardInfo shardInfo, boolean deleted) {
+        this.id = id;
+        this.shardId = shardId;
+        this.shardInfo = shardInfo;
+        this.deleted = deleted;
+    }
+
+    /**
+     * Returns the document id
+     * @return the id
+     */
+    public String id() {
+        return id;
+    }
+
+    /**
+     * Returns the shard id.
+     * @return the shard id, or a generated id if shards are not applicable
+     */
+    public ShardId shardId() {
+        return shardId;
+    }
+
+    /**
+     * Returns the shard info.
+     * @return the shard info, or generated info if shards are not applicable
+     */
+    public ShardInfo shardInfo() {
+        return shardInfo;
+    }
+
+    /**
+     * Returns whether deletion was successful
+     * @return true if deletion was successful, false if the object was not found
+     */
+    public boolean deleted() {
+        return deleted;
+    }
+
+    /**
+     * Class for constructing a Builder for this Response Object
+     */
+    public static class Builder {
+        private String id = null;
+        private ShardId shardId = null;
+        private ShardInfo shardInfo = null;
+        private boolean deleted = false;
+
+        /**
+         * Empty Constructor for the Builder object
+         */
+        public Builder() {}
+
+        /**
+         * Add an id to this builder
+         * @param id the id to add
+         * @return the updated builder
+         */
+        public Builder id(String id) {
+            this.id = id;
+            return this;
+        }
+
+        /**
+         * Adds a shard id to this builder
+         * @param shardId the shard id to add
+         * @return the updated builder
+         */
+        public Builder shardId(ShardId shardId) {
+            this.shardId = shardId;
+            return this;
+        }
+
+        /**
+         * Adds a generated shard id to this builder
+         * @param indexName the index name to generate a shard id
+         * @return the updated builder
+         */
+        public Builder shardId(String indexName) {
+            this.shardId = new ShardId(indexName, Strings.UNKNOWN_UUID_VALUE, 0);
```

Comment:
Answered above and also [here](https://github.com/opensearch-project/ml-commons/pull/2430#discussion_r1603993547), but I"ll repeat:

The `DeleteResponse` object requires a non-null ShardId.
```java
this.shardId = Objects.requireNonNull(shardId);
```
When there isn't one provided (such as DynamoDb) this code creates a simulated one.

Reply:
Yes, I saw your answer earlier too. I think for DDB / other non OS cases, if we just share a random shardID, that could be confusing for customers. I would prefer sharing any static String, maybe like Not Applicable or N/A, which can convey the actual meaning rather than causing unnecessary confusion.

Reply:
> I would prefer sharing any static String, maybe like `Not Applicable` or `N/A`, which can convey the actual meaning rather than causing unnecessary confusion.

The `Strings.UNKNOWN_UUID_VALUE` is `_na_`.